integrations/reporting_channels/gmail/gmail.py

```diff
@@ -148,16 +159,29 @@ def copy(self, uid, to_mailbox, from_mailbox=None):
         self.imap.uid('COPY', uid, to_mailbox)
 
     def fetch_multiple_messages(self, messages):
-        fetch_str =  ','.join(messages.keys())
+        if not isinstance(messages, dict):
+            raise Exception('Messages must be a dictionary')
+
+        fetch_str = ','.join(messages.keys())
         response, results = self.imap.uid('FETCH', fetch_str, '(BODY.PEEK[] FLAGS X-GM-THRID X-GM-MSGID X-GM-LABELS)')
-        for index in xrange(len(results) - 1):
-            raw_message = results[index]
-            if re.search(r'UID (\d+)', raw_message[0]):
-                uid = re.search(r'UID (\d+)', raw_message[0]).groups(1)[0]
-                messages[uid].parse(raw_message)
 
-        return messages
+        for raw_message in results:
+            if isinstance(raw_message, tuple):
+                uid_match = re.search(rb'UID (\d+)', raw_message[0])
+                if uid_match:
+                    uid = uid_match.group(1).decode('utf-8')
+                    if uid in messages:
+                        messages[uid].parse(raw_message)
+                    else:
+                        logging.warning(f'UID {uid} not found in messages dictionary')
+                else:
+                    logging.warning('UID not found in raw message')
+            elif isinstance(raw_message, bytes):
+                continue
```

Comment:
In case raw_message is byte why are we continuing here?

Reply:
Initial code part was having bytes, so the fetching data was failing since it is a tuple, removed the unnecessary logging and bytes as well.